File: templates/datasets/load.R
{{.Name}}.load <- function() {
    script.dir = getSrcDirectory(function(x) {x})
    destfile = file.path(script.dir, "{{.RelPath}}")

    data = data.frame()
    if (file.exists(destfile)) {
        data = read.csv(file=destfile)
    } else {
        stop("Dataset missing from disk")
    }

    return(data)
}

File: templates/datasets/load.py
import pandas as pd
import os


def load():
    """Reads csv file if it exists, otherwise raises an exception"""
    this_dir = os.path.dirname(os.path.realpath(__file__))
    destfile = os.path.join(this_dir, "{{.RelPath}}")

    if (os.path.isfile(destfile)):
        data = pd.read_csv(destfile)
    else:
        raise Exception("Dataset missing from disk")

    return data
